testsuite/src/test/java/org/ironjacamar/common/metadata/common/AbstractParserTestCase.java

```diff
@@ -0,0 +1,194 @@
+   {
+      assertEquals(expected, AbstractParser.transformExpression(expression));
+   }
+
+   /**
+    * Clear system properties set by tests
+    */
+   @After
+   public void tearDown()
+   {
+      System.clearProperty("Prop3");
+      System.clearProperty("Prop4");
+   }
+   
+   /**
+    * Simply test
+    */
+   @Test
+   public void testSimple()
+   {
+      exressionTransformTest("${Prop1:Property2}", "Property2");
+   }
+
+   /**
+    * Set the system property and  test
+    */
+   @Test
+   public void testWithPropertySet()
+   {
+      System.setProperty("Prop3", "1");
+      exressionTransformTest("${Prop3:Property1}", "1");
+   }
+
+   /**
+    * Test an incorrect expression
+    */
+   @Test
+   public void testWithIncorrectExpression()
+   {
+      exressionTransformTest("${Prop2:Property1", "${Prop2:Property1");
+   }
+
+   /**
+    * Test a complex expression
+    */
+   @Test
+   public void testComplexExpression()
+   {
+      exressionTransformTest("a${Prop1:Property1}_${Prop3:3}b", "aProperty1_3b");
+   }
+
+   /**
+    * Test a complex expression with properties set
+    */
+   @Test
+   public void testComplexExpressionWithProperties()
+   {
+      System.setProperty("Prop3", "4");
+      System.setProperty("Prop4", "5");
+      exressionTransformTest("a${Prop3:Property1}_${Prop4:3}b", "a4_5b");
+   }
+
+   /**
+    * Test an expression without a default value
+    */
+   @Test
+   public void testWithoutDefaultValue()
+   {
+      exressionTransformTest("${Prop1}", "");
+   }
+
+   /**
+    * Test an expression without a default value. Property is set.
+    */
+   @Test
+   public void testWithoutDefaultValuePropertySet()
+   {
+      System.setProperty("Prop3", "6");
+      exressionTransformTest("${Prop3}", "6");
+   }
+
+   /**
+    * Test a complex expression without a default value
+    */
+   @Test
+   public void testComplexWithoutDefaultValue()
+   {
+      exressionTransformTest("1${Prop1}2${Prop3}4", "124");
+   }
+
+   /**
+    * Test a complex expression without a default value. Property is set.
+    */
+   @Test
+   public void testComplexWithoutDefaultValuePropertySet()
+   {
+      System.setProperty("Prop3", "4");
+      System.setProperty("Prop4", "5");
+      exressionTransformTest("a${Prop3}_${Prop4}b", "a4_5b");
+   }
+
+   /**
+    * Test a complex expression with mixed properties.
+    */
+   @Test
+   public void testComplexMixedProperties()
+   {
+      System.setProperty("Prop3", "8");
+      exressionTransformTest("a${Prop3:6}_${Prop9}7:b", "a8_7:b");
+   }
+
+   /**
+    * Test special symbols substituted by File constants
+    */
+   @Test
+   public void testSpecialCases()
+   {
+      exressionTransformTest("a${:}_${/}b", "a" + File.pathSeparator + "_" + File.separator + "b");
+   }
+
+   /**
+    * Test nested expressions
+    */
+   @Test
+   public void testNestedExpressions()
+   {
+      exressionTransformTest("${Prop1:${Prop2:default}1}", "default1");
```

Comment:
That is not a valid expression - it should be ${Prop1:${Prop2:default}} -> default